subsystem/kubeapi_test.go

```diff
@@ -834,6 +834,38 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		Expect(cluster.ImageGenerated).Should(Equal(true))
 	})
 
+	It("deploy infraEnv before clusterDeployment", func() {
+		infraEnvName := "infraenv"
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		infraEnvSpec := getDefaultInfraEnvSpec(secretRef, clusterDeploymentSpec)
+
+		deployInfraEnvCRD(ctx, kubeClient, infraEnvName, infraEnvSpec)
+		infraEnvKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      infraEnvName,
+		}
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
+		configureLocalAgentClient(cluster.ID.String())
+
+		// InfraEnv Reconcile takes longer, since it needs to generate the image.
+		Eventually(func() string {
+			condition := conditionsv1.FindStatusCondition(getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.Conditions, v1beta1.ImageCreatedCondition)
+			if condition != nil {
+				return condition.Message
+			}
+			return ""
+		}, "2m", "2s").Should(Equal(v1beta1.ImageStateCreated))
```

Comment:
Move to a function `checkInfraEnvCreated(name string)` that will be used by the other tests as well.

Reply:
See #1570

Reply:
Done